outputs.tf

```diff
@@ -1,6 +1,11 @@
 output "resource" {
   value       = azurerm_key_vault.this
   description = "The Key Vault resource."
+
+  depends_on = [
+    azurerm_key_vault_key.this,
```

Comment:
This would be implied because of the value, I think we can remove this line.

Reply:
Hi Matt. It is needed as the VM takes a dependency on the keyvault id. So without that depends_on the VM creation would proceed without the RBAC being assigned and key stored in the KV. This ends-up with the permission error described in the issue.
By adding the depends on it will ensure the rbac has been set, key has been stored and then return the resource output attribute. That issue was hidden because the vm example I reused had an explicit depends_on, but I would prefer adding the depends_on in the output resource.